packages/design-system/src/components/swatch/swatch.js

```diff
@@ -168,7 +168,14 @@ function Swatch({
 
 Swatch.propTypes = {
   children: PropTypes.node,
-  pattern: PropTypes.object,
+  pattern: PropTypes.shape({
```

Comment:
Use `HexPropType` from the `@web-stories-wp/patterns` package.

Maybe create a new `ColorPropType` there:

```js
export const ColorPropType = PropTypes.shape({
  color: HexPropType.isRequired,
});
```